lib/devise/custom_auth_failure.rb

```diff
@@ -7,9 +7,9 @@
 class CustomAuthFailure < Devise::FailureApp
   def respond
     if scope == :hmis_user
-      return redirect_to_hmis if ENV['HMIS_OKTA_CLIENT_ID'].present?
+      return json_error_response if request.content_type == 'application/json'
```

Comment:
Are you clear on the difference between these two 👀 ? (Because I'm not).

[This test failure](https://github.com/greenriver/hmis-warehouse/actions/runs/6727713787/job/18285931400) occurs when checking for `request.content_type == 'application/json'` but not when checking for `request.format == :json`.


```suggestion
      return json_error_response if request.content_type == 'application/json' || request.format == :json
```

Reply:
Yeah it is very odd, I don't understand it either. I suspect request.format works for the devise routes but not for the graphql API. But it also might be related to the test env, the request might be have different behavior there.

In any case it seems like a bit of a rabbit hole, let's go with your fix. Thank you!!

Reply:
Alright, sounds good, this seems safe to me. Thanks!